base/math.jl

```diff
@@ -536,38 +536,48 @@ Stacktrace:
 [...]
 ```
 """
-hypot(x::Number, y::Number) = hypot(promote(x, y)...)
-function hypot(x::T, y::T) where T<:Number
-    ax = abs(x)
-    ay = abs(y)
-    if ax < ay
-        ax, ay = ay, ax
-    end
+function hypot(x::Number, y::Number)
+	ay, ax = minmax(abs(x), abs(y))
     if iszero(ax)
         r = ay / oneunit(ax)
     else
         r = ay / ax
     end
-
     rr = ax * sqrt(1 + r * r)
 
-    # Use type of rr to make sure that return type is the same for
-    # all branches
-    if isnan(r)
-        isinf(ax) && return oftype(rr, Inf)
-        isinf(ay) && return oftype(rr, Inf)
-        return oftype(rr, r)
-    else
-        return rr
-    end
+	if isnan(r)
+		if isinf(x) || isinf(y)
+			return oftype(rr, Inf)
+		else
+			return oftype(rr, NaN)
+		end
+   	else
+   		return rr
+   	end
 end
-
+#= Specialized one and two argument versions are
+faster than generic varargs version below =#
+hypot(x::Number) = abs(float(x))
 """
     hypot(x...)
 
 Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 """
-hypot(x::Number...) = sqrt(sum(abs2(y) for y in x))
+hypot(x::Number, y::Number...) = _hypot(promote(x, y...))
```

Comment:
What's the motivation for this? If this is to disambiguate the 1-arg method (is that really needed?), then it doesn't seem worthwhile: calling `hypot` with a single argument should be an extremely rare occurrence; imo, it doesn't seem worth it to complicate this method further for such a rare thing (even if a slightly faster method exists)

Reply:
It's not unreasonable to me just to define `hypot() = 0`.

Reply:
@stevengj Having `hypot()` seems inconsistent with other functions such as `max`, `min`, which err with no arguments. The previous version of `hypot` (before this PR) also errors without arguments. Also, why `0::Int` is the correct answer? Why not `0.0`, `false`, `Int16(0)`?

I think IEEE doesn't help here since it only specifies the two argument version?

Reply:
Alternatively, the `hypot()` call corresponds to an `N = 0` case; since `N` is a known type parameter, that case could be inferred and treated separately statically: i.e., it shouldn't cost anything to test `N == 0 && error( ... )` inside the function - the test would be removed at compilation for `N != 0`, AFAIK.

Reply:
I'm less worried about `0::Int` (since numeric zeros interoperate) than someone extending `hypot` for non-`Number` types, or using it for dimensionful types, in which case `0` is not an additive identity.

Reply:
@thchr I removed the one arg version and did the `N==0` error case as you suggested